src/haz3lcore/dynamics/DH.re

```diff
@@ -1,52 +1,55 @@
 open Sexplib.Std;
+open Hazel_menhir.AST;
 
 [@deriving (show({with_path: false}), sexp, yojson)]
 type if_consistency =
   | ConsistentIf
   | InconsistentIf;
 
 module rec DHExp: {
+  //All comments show the textual syntax for DHExp when built using the menhir parser
   [@deriving (show({with_path: false}), sexp, yojson)]
   type t =
-    | EmptyHole(MetaVar.t, HoleInstanceId.t)
-    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t)
-    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t)
+    | EmptyHole(MetaVar.t, HoleInstanceId.t) //()
+    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t) //(_HOLE x)
+    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t) //(_FREE x)
     | InvalidText(MetaVar.t, HoleInstanceId.t, string)
-    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case)
+    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case) //(_HOLE (case ...))
     | Closure([@opaque] ClosureEnvironment.t, t)
-    | Filter(DHFilter.t, t)
-    | BoundVar(Var.t)
-    | Sequence(t, t)
-    | Let(DHPat.t, t, t)
-    | FixF(Var.t, Typ.t, t)
-    | Fun(DHPat.t, Typ.t, t, option(Var.t))
+    | Filter(DHFilter.t, t) //pause a == 2 b
+    | BoundVar(Var.t) //x
+    | Sequence(t, t) //a; b
+    | Let(DHPat.t, t, t) //let x = y in z
+    | FixF(Var.t, Typ.t, t) //_FIX f Int -> Int fun: Int x -> 1 + x f in 55
```

Comment:
I believe the syntax here should be `fix x : ty -> e`

Reply:
Ok, I fixed that but I unfortunately cant use the "->" because it seems to create a conflict in Menhir between the exp and typ (I'm pretty sure it is conflicting with the "->" in arrow types) so instead I used "=>". The syntax is now `fix x: ty => e`, are you alright with that or would you prefer I try to find some way around the conflict?

Reply:
Maybe have it be parenthesized, so `fix x -> e` or `fix (x : ty) -> e`? would prefer to keep the Hazel notation consistent here so if we change it here then we should change it in Hazel

Reply:
That worked! Thank you!